vnext/Microsoft.ReactNative/Views/SwitchViewManager.cpp

```diff
@@ -66,21 +110,32 @@ void SwitchShadowNode::UpdateThumbColor() {
     return;
 
   const auto thumbBrush = IsValidColorValue(m_thumbColor) ? BrushFrom(m_thumbColor) : nullptr;
-  UpdateToggleSwitchThumbResourceBrushes(toggleSwitch, thumbBrush);
+  if (thumbBrush != nullptr) {
+    UpdateToggleSwitchThumbResourceBrushes(toggleSwitch, thumbBrush);
+  }
 }
 
-void SwitchShadowNode::UpdateTrackColor() {
+void SwitchShadowNode::UpdateTrackColorOn() {
   auto toggleSwitch = GetView().as<winrt::ToggleSwitch>();
   if (toggleSwitch == nullptr)
     return;
 
   const auto onTrackBrush = IsValidColorValue(m_onTrackColor) ? BrushFrom(m_onTrackColor) : nullptr;
+  UpdateToggleSwitchTrackResourceBrushesOn(toggleSwitch, onTrackBrush);
+}
+
+void SwitchShadowNode::UpdateTrackColorOff() {
+  auto toggleSwitch = GetView().as<winrt::ToggleSwitch>();
+  if (toggleSwitch == nullptr)
+    return;
```

Comment:
I'm not sure what this is trying to do - the view always is a toggleswitch... is this tracking whether the view was destroyed already, or? I see the same code above, it's probably wrong there too
`.as<T>()` throws if the object cannot be QI'd for T, whereas `.try_as<T>()` will return nullptr if it cannot be QI'd for T

Reply:
yeah! I left that in since it was there before me but that makes sense! Looking at the blame, that was added in 2 years ago and just never removed. I'll remove it now